src/ui/component/dayPicker/dayPicker.tsx

```diff
@@ -39,20 +43,78 @@ export const DayPicker: FC<DayPickerProps> = ({
 
   const getLocaleFromLocaleString = useMemo(() => localeMap.get(locale) ?? enUS, [locale])
 
+  const disabledDays = useMemo(() => {
+    const startOfFromYear = new Date(fromYear, 0, 1)
+    const endOfToYear = new Date(toYear, 11, 31)
+
+    switch (type) {
+      case 'rangeFrom':
+        return toDate ? [{ after: toDate, before: endOfToYear }] : []
+      case 'rangeTo':
+        return fromDate ? [{ after: startOfFromYear, before: fromDate }] : []
+      case 'single':
+        return []
+    }
+  }, [type, fromDate, toDate, fromYear, toYear])
+
+  const middleRange = useMemo(() => {
+    if (!fromDate || !toDate || !isAfter(subDays(toDate, 1), fromDate)) {
+      return []
+    }
+
+    const middleFrom = addDays(fromDate, 1)
+    const middleTo = subDays(toDate, 1)
+    return [{ from: middleFrom, to: middleTo }]
+  }, [fromDate, toDate])
+
+  const startRange = useMemo(() => {
+    if (!fromDate || !toDate || isSameDay(fromDate, toDate)) {
+      return []
+    }
+
+    return [fromDate]
+  }, [fromDate, toDate])
+
+  const endRange = useMemo(() => {
+    if (!fromDate || !toDate || isSameDay(fromDate, toDate)) {
+      return []
+    }
+
+    return [toDate]
+  }, [fromDate, toDate])
+
+  const selectedDayFromOtherPicker = useMemo(() => {
```

Comment:
What this modifier means?

Reply:
The name selectedDayFromOtherPicker is confusing because it implies a relationship with another picker component, which is not clear here. A better name would focus on the role or function that the modifier performs.

This modifier highlights the boundary date of a dateRange.
We can rename the modifier with `rangeEndpoint` for instance.

What do you think? Thanks

Reply:
Yes, sounds better, thanks